### PR TITLE
feat: add uuid, date, date-time, uri format validators

### PR DESCRIPTION
**Pull request by jarvis9443**
Add PCRE regex patterns for `uuid`, `date`, `date-time`, and `uri` formats to the `reg_map` table, enabling format validation for these commonly used JSON Schema formats.

**Changes:**
- `lib/jsonschema.lua`: added 4 new format regex entries to `reg_map`, updated error message from "address" to "format"
- `spec/extra/format.json`: added test cases for each new format (valid/invalid/non-string inputs, plus URI edge cases for whitespace and empty path)
- `.github/workflows/ci.yml`: updated runner from ubuntu-18.04 to ubuntu-22.04, modernized actions and GPG key handling

All existing tests pass locally.

## Summary by CodeRabbit

* **New Features**
  * Added validation for UUID, date, date-time, and URI formats with clearer format-failure messages.

* **Tests**
  * Added comprehensive tests covering valid and invalid UUID, date, date-time, and URI cases, including variants and edge cases.

* **Chores**
  * Modernized CI workflow environment and package installation steps to update build tooling and dependencies.

### CODE REVIEW

File: lib/jsonschema.lua [hunk 1]
```diff
@@ -586,7 +586,11 @@ local reg_map = {
...
-  ["hostname"] = [[^[a-zA-Z0-9\-\.]+$]]
+  ["hostname"] = [[^[a-zA-Z0-9\-\.]+$]],
+  ["uuid"] = [[^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$]],
+  ["date"] = [[^\d{4}-(?:0[1-9]|1[0-2])-(?:0[1-9]|[12]\d|3[01])$]],
+  ["date-time"] = [[^\d{4}-(?:0[1-9]|1[0-2])-(?:0[1-9]|[12]\d|3[01])[Tt](?:[01]\d|2[0-3]):[0-5]\d:[0-5]\d(?:\.\d+)?(?:[Zz]|[+-](?:[01]\d|2[0-3]):[0-5]\d)$]],
+  ["uri"] = [[^[a-zA-Z][a-zA-Z0-9+\-.]*:.+$]],
```
[the remark this answers is not in the document]
Comment (jarvis9443) [reply lib/jsonschema.lua]: Good catch — updated the URI regex to `[^\s]*` to reject whitespace and accept empty-path URIs like `foo:`. Also added test cases for both edge cases.

File: spec/extra/format.json [hunk 2]
```diff
@@ -107,5 +107,179 @@
...
+            "description": "invalid URI: scheme only no colon",
+            "data": "http",
+            "valid": false
+        },
+        {
+            "description": "non-string is valid",
+            "data": 123,
+            "valid": true
+        }
+    ]
```
[the remark this answers is not in the document]
Comment (jarvis9443) [reply spec/extra/format.json:293]: Added both test cases — `foo:` as valid and whitespace URI as invalid.

File: spec/extra/format.json [hunk 3]
```diff
@@ -107,5 +107,179 @@
...
+            "description": "invalid date: wrong format",
+            "data": "01-15-2024",
+            "valid": false
+        },
+        {
+            "description": "non-string is valid",
+            "data": 20240115,
+            "valid": true
+        }
+    ]
```
[the remark this answers is not in the document]
Comment (jarvis9443) [reply spec/extra/format.json:186]: These calendar-edge cases (`2024-02-31`, `2023-02-29`) would require semantic validation beyond regex, which is out of scope for this PR. The regex validates the format structure per RFC 3339 syntax. See the reply on the date regex comment above.

File: lib/jsonschema.lua [hunk 4]
```diff
@@ -586,7 +586,11 @@ local reg_map = {
...
-  ["hostname"] = [[^[a-zA-Z0-9\-\.]+$]]
+  ["hostname"] = [[^[a-zA-Z0-9\-\.]+$]],
+  ["uuid"] = [[^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$]],
+  ["date"] = [[^\d{4}-(?:0[1-9]|1[0-2])-(?:0[1-9]|[12]\d|3[01])$]],
+  ["date-time"] = [[^\d{4}-(?:0[1-9]|1[0-2])-(?:0[1-9]|[12]\d|3[01])[Tt](?:[01]\d|2[0-3]):[0-5]\d:[0-5]\d(?:\.\d+)?(?:[Zz]|[+-](?:[01]\d|2[0-3]):[0-5]\d)$]],
```
[the remark this answers is not in the document]
Comment (jarvis9443) [reply lib/jsonschema.lua:592]: Valid point, but adding semantic calendar validation (leap year, days per month) would require changing the code generation approach — similar to how ipv4/ipv6 use custom parse functions with `inet_pton`. For now the regex provides format-level validation which is what most JSON Schema validators do for `date` / `date-time`. Full calendar validation could be a follow-up if needed.

**Review by nic-6443 — APPROVED** [review]

File: spec/extra/format.json [hunk 5]
```diff
@@ -107,5 +107,189 @@
             "valid": false
         }
     ]
+  },
+  {
+    "description": "validation of UUID",
```
Comment (nic-6443) [inline spec/extra/format.json:112]: We need to update the GitHub workflow for this repository. It uses an outdated runner and cannot run now: https://github.com/api7/jsonschema/actions/runs/24717002603/job/72295908024?pr=96

Comment (jarvis9443) [reply spec/extra/format.json:112]: Fixed — updated CI to ubuntu-22.04, actions/checkout@v4, leafo/gh-actions-lua@v10, and switched from deprecated apt-key to gpg keyring for OpenResty repo.

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -29,15 +29,14 @@ jobs:
           luarocksVersion: "3.8.0"
 
       - name: Linux Get dependencies
-        run: sudo apt install -y build-essential libncurses5-dev libreadline-dev libssl-dev perl
+        run: sudo apt-get install -y build-essential libncurses5-dev libreadline-dev libssl-dev perl libpcre3-dev
```
[the remark this answers is not in the document]
Comment (nic-6443) [reply .github/workflows/ci.yml]: @jarvis9443 This needs to be handled.

[the remark this answers is not in the document]
Comment (jarvis9443) [reply .github/workflows/ci.yml]: Fixed — added `apt-get update` before the first package install step.
